torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -258,6 +267,37 @@ struct KinetoThreadLocalState : public ProfilerThreadLocalState {
         activity.addMetadata(
             "Sequence number",
             std::to_string(kineto_event.sequenceNr()));
+
+        if (kineto_event.fwdThreadId() > 0) {
```

Comment:
Can you move this code into its own function, e.g. generateForwardBackwardLink(activity, tidSeq2activity, fwd_bwd_link_id)?

Reply:
Thanks. Done.